packages/webapp/src/components/Forms/SimpleCustomType/index.jsx

```diff
@@ -0,0 +1,99 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useForm } from 'react-hook-form';
+import Form from '../../Form';
+import PageTitle from '../../PageTitle/v2';
+import Input, { getInputErrors } from '../../Form/Input';
+import Button from '../../Form/Button';
+import PropTypes from 'prop-types';
+
+// onSubmit: should be used in Add, Edit view
+// onClick: should be used in Read-only view
+// defaultValue: should be used in Read-only, Edit view
+const PureSimpleCustomType = ({
+  handleGoBack,
+  onSubmit,
+  onClick,
+  view,
+  buttonText,
+  pageTitle,
+  inputLabel,
+  customTypeRegister,
+  defaultValue,
+}) => {
+  const { t } = useTranslation();
+
+  const {
+    handleSubmit,
+    register,
+    formState: { errors, isValid },
+  } = useForm({
+    mode: 'onChange',
+    defaultValues: { [customTypeRegister]: defaultValue || undefined },
+  });
+  const MAX_CHARS = 100;
```

Comment:
Since we plan to [refactor](https://lite-farm.atlassian.net/browse/LF-3571) the task type, you could make `MAX_CHARS` a prop. (the current limit for task type is 25, so it will need to be passed from its parent component)

Reply:
@SayakaOno do you think it would be best if we standardised this? So instead just allowing 100 characters for Tasktype moving forward?

Reply:
In the future, maybe... (we should ask David) but I imagine it would be refactoring + changing the limit (we would need extra work like checking the backend etc), so my approach would be focus on just refactoring when working on the ticket.